nibabel/openers.py

```diff
@@ -145,3 +145,13 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close_if_mine()
+
+    @classmethod
+    def register_extension(opener_klass, ext, func):
+        """Decorator"""
+        def decorate(klass):
+            assert ext not in opener_klass.compress_ext_map, \
+                "Cannot redefine extension-function mappings."
+            opener_klass.compress_ext_map[ext] = func
```

Comment:
For example, we only need the `.mgz` compressed extension to be in play for freesurfer files.  So maybe we can get away with having a `MgzOpener` class that is the same as  current `BinOpener`, and putting the class (`Opener`) into `ArrayProxy` and `FileHolder` classes as class attributes.  Then the MGH format can subclass these two objects to add the `MgzOpener` class?